doc/user-guide.md

```diff
@@ -400,6 +400,70 @@ func main() {
 
 After adding new import paths to your operator project, run `dep ensure` in the root of your project directory to fulfill these dependencies.
 
+## Leader election
+
+For improved reliability and quick failover it may be necessary to run multiple replicas of the operator with leader election, so that one leader instance handles the reconciliation while the other instances are inactive but ready to take over when the leader fails.
+
+There are two different leader election implementations to choose from, each with its own tradeoff.
+
+- [Leader-for-life][leader_for_life]: The leader pod only gives up leadership (via garbage collection) when it is deleted. Avoids the possibility of 2 instances mistakenly running as leaders(split brain). However, this method can be slow to elect a new leader when the leader pod is on an unresponsive node(partitioned) where it takes a long time for the leader pod to be deleted.
```

Comment:
Rather than use words like "slow", which is relative and subjective, I suggest referencing the node failure timings. I don't remember the name of the setting off the top of my head, but there is a configurable time frame (default 5 minutes) after which workloads on an unresponsive node get deleted. Referencing that setting here will also give people an opportunity to consider modifying it. I'll try to look it up later on.

I'm learning more about node lifecycle, and my suspicion is that in practice the period of uncertainty will normally not last for 5 minutes. Using the `cloud-controller-manager`, the status of the underlying machine can be queried with the cloud environment (AWS, azure, openstack, whatever), and a dead machine can have its node deleted more promptly. So I suspect in real life this won't be a common scenario.

Reply:
I believe that's the `pod-eviction-timeout`. From the [node architecture docs](https://kubernetes.io/docs/concepts/architecture/nodes/#condition):
> If the Status of the Ready condition remains Unknown or False for longer than the pod-eviction-timeout, an argument is passed to the kube-controller-manager and all the Pods on the node are scheduled for deletion by the Node Controller. The default eviction timeout duration is five minutes

I'll make this more specific by pointing out the default time for pod eviction that can result in an unresponsive leader for the given period of time.